webapp-django/crashstats/api/tests/test_views.py

```diff
@@ -86,6 +86,15 @@ def test_invalid_url(self):
         response = self.client.get(url)
         eq_(response.status_code, 404)
 
+    def test_base_classes_raise_bad_request(self):
+        url = reverse('api:model_wrapper', args=('SocorroMiddleware',))
+        response = self.client.get(url)
+        eq_(response.status_code, 400)
```

Comment:
Should be 404. 400 is when the URL is valid but there's something wrong with the parameters. 
Equally, if you do `curl .../api/cm49b7aktgja7v` that should definitely be a 404.